source/common/router/router.cc

```diff
@@ -594,7 +596,9 @@ void Filter::onUpstreamReset(UpstreamResetType type,
           streamResetReasonToResponseFlag(reset_reason.value());
       callbacks_->streamInfo().setResponseFlag(response_flags);
       code = Http::Code::ServiceUnavailable;
-      body = "upstream connect error or disconnect/reset before headers";
+      body =
+          absl::StrCat("upstream connect error or disconnect/reset before headers. reset reason: ",
```

Comment:
I think this is a good change. The only thing that gives me pause is that this terrible error message is the subject of so many support queries, etc. that I worry someone out there has baked it into their code to expect it. Personally, I'm fine changing it. @envoyproxy/maintainers any thoughts here?

Reply:
I thought about this being baked in to scripts/code but since this is documented behaviour I think we should be OK to change - but open to other suggestions

Reply:
+1 to adding more detail.  This doesn't hit my threshold for runtime guarding (though if anyone has serious complaints we can backport runtime workarounds once the singleton PR lands as a test of "isn't this easy now?" :-)  

If we're worried about breakage we should probably note it in the version history and if we're really worried we could envoy-announce.

Reply:
+1, let's add a release note and call it good?

/wait